loadtest/control/simplecontroller/actions.go

```diff
@@ -108,3 +108,87 @@ func (c *SimpleController) viewChannel() control.UserStatus {
 		return control.UserStatus{User: c.user, Info: fmt.Sprintf("channel viewed. result: %v", channelViewResponse.ToJson())}
 	*/
 }
+
+func (c *SimpleController) reload() control.UserStatus {
+	// Getting preferences.
+	err := c.user.GetPreferences()
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+
+	prefs, _ := c.user.Store().Preferences()
+	userIds := make([]string, len(prefs))
+	chanId := ""
+	for i, p := range prefs {
+		if p.Name == model.PREFERENCE_NAME_LAST_CHANNEL {
+			chanId = p.Value
+		}
+		userIds[i] = p.UserId
+	}
```

Comment:
There's a slight problem with this. The `p.UserId` we are using is always going to be the current user's id.  
What we are looking for here could be, for example, the users with whom we have direct channels open.  
To find that info we need to look for `p.Category == PREFERENCE_CATEGORY_DIRECT_CHANNEL_SHOW` and then for `p.Name` which should hold that user's id.  
If this doesn't make 100% sense I agree but it looks that's how the API is currently working.

Reply:
Yeah that's right actually. I am not fully versed with the APIs, so I will defer to your judgement. :+1: